codelabs/grpc-java-opentelemetry/start_here/src/main/java/io/grpc/codelabs/opentelemetry/OpenTelemetryServer.java

```diff
@@ -86,7 +86,24 @@ public static void main(String[] args) throws IOException, InterruptedException
     }
 
     ///////////////////////////////////////////////////////////////////////////
-    // CODELAB HINT : Add code to register GrpcOpenTelemetry here.
+    // CODELAB HINT : Add code to create Prometheus exporter here. The default
+    // port has been initialized for you in a variable i.e prometheusPort.
+    // Please use that while configuring teh exporter.
+    ///////////////////////////////////////////////////////////////////////////
+
+    ///////////////////////////////////////////////////////////////////////////
+    // CODELAB HINT : Add code for the following tasks here:
+    // 1. After creating PrometheusHttpServer instance, use that
+    // as a MetricReader to create SdkMeterProvider.
```

Comment:
Is it a MetricReader or MetricWriter?

Reply:
It is a [Metric Reader](https://javadoc.io/doc/io.opentelemetry/opentelemetry-sdk-metrics/latest/io/opentelemetry/sdk/metrics/SdkMeterProviderBuilder.html#registerMetricReader), which reads metrics produced by instrumentation.